web_external/js/models/DatasetModel.js

```diff
@@ -66,313 +39,153 @@ minerva.models.DatasetModel = minerva.models.MinervaModel.extend({
     },
 
     getDatasetType: function () {
-        // this is the start of trying to build an interface around the minerva metadata
         var minervaMetadata = this.metadata();
         return _.has(minervaMetadata, 'dataset_type') ? minervaMetadata.dataset_type :
             (_.has(minervaMetadata, 'original_type') ? minervaMetadata.original_type : null);
     },
 
-    // functions dealing with json type
-    // TODO split out to a subclass
-
-    getJsonRow: function () {
-        var minervaMetadata = this.metadata();
-        if (!_.has(minervaMetadata, 'json_row')) {
-            girder.restRequest({
-                path: 'minerva_dataset/' + this.get('_id') + '/jsonrow',
-                type: 'POST'
-            }).done(_.bind(function (resp) {
-                this.metadata(resp);
-                this.trigger('m:jsonrowGot', this);
-            }, this)).error(_.bind(function (err) {
-                console.error(err);
-                girder.events.trigger('g:alert', {
-                    icon: 'cancel',
-                    text: 'Could not get jsonrow in dataset item.',
-                    type: 'error',
-                    timeout: 4000
-                });
-            }, this));
-        } else {
-            this.trigger('m:jsonrowGot', this);
+    ////////////////////////////////////////////////////////
+    // GeoJs rendering api.                               //
+    ////////////////////////////////////////////////////////
+
+    /**
+     * Initialize the GeoJs rendering type along with any secondary data needed for
+     * GeoJs rendering, inferring the rendering type based on
+     * the dataset's metadata, and saving the dataset's metadata if changed.
+     *
+     * @returns {Object} Updated minerva metadata of this dataset.
+     */
+    _initGeoRender: function () {
```

Comment:
It seems like this is something that should be part of the models initialize method, no? That way the rest of the methods can assume it's loaded already.